src/energyplus/Test/HeatPumpPlantLoopEIR_GTest.cpp

```diff
@@ -0,0 +1,96 @@
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "EnergyPlusFixture.hpp"
+
+#include "../ForwardTranslator.hpp"
+
+#include "../../model/Model.hpp"
+#include "../../model/HeatPumpPlantLoopEIRCooling.hpp"
+#include "../../model/HeatPumpPlantLoopEIRHeating.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/PlantLoop.hpp"
+
+#include <utilities/idd/HeatPump_PlantLoop_EIR_Cooling_FieldEnums.hxx>
+#include <utilities/idd/HeatPump_PlantLoop_EIR_Heating_FieldEnums.hxx>
+
+#include <utilities/idd/IddEnums.hxx>
+#include "../../utilities/idf/IdfObject.hpp"
+#include "../../utilities/idf/IdfObject_Impl.hpp"
+
+#include "../../utilities/idf/WorkspaceObject.hpp"
+#include "../../utilities/idf/WorkspaceObject_Impl.hpp"
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_HeatPumpPlantLoopEIR) {
+
+  Model m;
+
+  PlantLoop plant_loop_cup_clg(m);
+  PlantLoop plant_loop_cup_htg(m);
+  PlantLoop plant_loop_plhp_clg(m);
+  PlantLoop plant_loop_plhp_htg(m);
+
+  HeatPumpPlantLoopEIRCooling plhp_clg(m);
+  HeatPumpPlantLoopEIRHeating plhp_htg(m);
+
+  EXPECT_TRUE(plant_loop_cup_clg.addDemandBranchForComponent(plhp_clg));
+  EXPECT_TRUE(plant_loop_plhp_clg.addSupplyBranchForComponent(plhp_clg));
+  EXPECT_TRUE(plant_loop_cup_htg.addDemandBranchForComponent(plhp_htg));
+  EXPECT_TRUE(plant_loop_plhp_htg.addSupplyBranchForComponent(plhp_htg));
+
+  // #3837: These two reference each other, and we want to avoid a recursion problem (each FT function calling each other)
+  EXPECT_TRUE(plhp_clg.setCompanionHeatingHeatPump(plhp_htg));
+  EXPECT_TRUE(plhp_htg.setCompanionCoolingHeatPump(plhp_clg));
+
+  openstudio::energyplus::ForwardTranslator ft;
+  Workspace w = ft.translateModel(m);
+
+  EXPECT_EQ(0u, ft.errors().size());
+
+  {
+    WorkspaceObjectVector idf_ccs(w.getObjectsByType(IddObjectType::HeatPump_PlantLoop_EIR_Cooling));
+    EXPECT_EQ(1u, idf_ccs.size());
+    WorkspaceObject idf_cc(idf_ccs[0]);
+    // Companion
+    EXPECT_EQ(plhp_htg.nameString(), idf_cc.getString(HeatPump_PlantLoop_EIR_CoolingFields::CompanionHeatPumpName).get());
+  }
+
+  {
+    WorkspaceObjectVector idf_hcs(w.getObjectsByType(IddObjectType::HeatPump_PlantLoop_EIR_Heating));
+    EXPECT_EQ(1u, idf_hcs.size());
+    WorkspaceObject idf_hc(idf_hcs[0]);
+    // Companion
+    EXPECT_EQ(plhp_clg.nameString(), idf_hc.getString(HeatPump_PlantLoop_EIR_HeatingFields::CompanionHeatPumpName).get());
+  }
```

Comment:
I would like to see:

* [ ] A test for all regular fields to make sure we find the same value as the setters you'll add above
* [ ] Node / branch tests to ensure you write the appropriate nodes to the Load / Source sides.
* [ ] A PlantEquipmentSchemes test, to ensure that the heating coil is on a PlantEqOpHeatingLoad for the heating load loop, and the cooling coil is on a PlantEqOpCoolingLoad for the cooling load loop.

Reply:
@jmarrec Looks like you took care of the second and third items from above?